TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -349,5 +349,50 @@ public LockTableStatement CreateLockTableStatement(CodeElementsParser.LockTableS
 
             return new LockTableStatement(tableName, partitionId, mode);
         }
+
+        public DropTableStatement CreateDropTableStatement(CodeElementsParser.DropTableStatementContext context)
+        {
+            if (context.tableOrAliasName() != null)
+            {
+                var name = (Token) context.tableOrAliasName().tableOrViewOrCorrelationName().Name;
+                var schemaName = (Token) context.tableOrAliasName().tableOrViewOrCorrelationName().SchemaName;
+                var dbms = (Token) context.tableOrAliasName().tableOrViewOrCorrelationName().DBMS;
+                var tableName = new TableViewCorrelationName(CreateTableOrAliasName(name, schemaName, dbms));
+                return new DropTableStatement(tableName);
+            }
+            return null;
+        }
+        public SymbolReference CreateTableOrAliasName (Token nameToken, Token qualifierToken,
```

Comment:
- private method
- instead of having the caller extract parts of the name, pass the tableOrAlias context and extract parts inside the method
- as for the AmbiguousSymbolReference: the reference is ambiguous on its type not because it's not qualified so tehmethod will always create an AmbiguousSymbolReference instance

Reply:
The ambiguity is between a non-qualified table name and an alias name. 
I think the method should return tableOrViewOrCorrelationName which is the type of tableOrAliasName. And :
- If the name is qualified , the tableOrAliasName will be composed of a QualifiedSymbolRef.
- If the name is not qualified we can't dfferenciate if it's a non-qualified table name or an alias name  so it will be composed of an  AmbiguousSymbolReference .